[firebase_auth] Ability to choose between sign-in or link on auto verification of phone number on Android

cqhchan opened the pull request:
- Fixed Android Link with credentials bug.
- Added in ability to choose between sign-in or link on auto verification of phone number

ened commented on packages/firebase_auth/lib/src/firebase_auth.dart:
```diff
@@ -241,15 +241,20 @@ class FirebaseAuth {
+      @required PhoneVerificationFailed verificationFailed,
+      @required PhoneCodeSent codeSent,
+      @required PhoneCodeAutoRetrievalTimeout codeAutoRetrievalTimeout,
+      bool linkCredentials = false}) async {
```
Could you please add back the comma (,) at the end of the array setup and reformat?

ened commented on packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:
```diff
@@ -173,24 +173,44 @@ private void handleVerifyPhoneNumber(
+                          if (task.isSuccessful()) {
+                            Map<String, Object> arguments = new HashMap<>();
+                            arguments.put("handle", handle);
+                            channel.invokeMethod("phoneVerificationCompleted", arguments);
```
Could the 2 onComplete handlers be merged?

cqhchan replied:
Hi @ened  completed the requested changes

ened commented on packages/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:
```diff
@@ -173,24 +173,44 @@ private void handleVerifyPhoneNumber(
+                  .linkWithCredential(phoneAuthCredential)
+                  .addOnCompleteListener(
+                      new OnCompleteListener<AuthResult>() {
+
```
Please remove empty lines.

cqhchan commented:
@ened  completed the requested changes

cqhchan commented:
@tolotrasamuel 

`
  firebase_auth:
    git:
      url: git://github.com/cqhchan/plugins.git
      path: packages/firebase_auth
`

Insert this into pubspec.yaml to use it.  

https://github.com/cqhchan/plugins

This makes use of credentials implementation in https://github.com/flutter/plugins/pull/928 
I fixed a few bugs with their implementation and added a bool value in 
`
Future<void> verifyPhoneNumber(...)
`
to allow you to differentiate between verifyPhoneNumber for Login or For linking credentials with existing account. This addition bool value only affects specific android devices where they have auto verification.

collinjackson commented:
Thanks for the contribution!

We've fixed the linkWithCredential issue (flutter/plugins#1333).

I'd love to get the other parts of your change, but I think we need an iOS implementation. Before we land this I'd ideally like to figure out how to write integration tests for phone auth so that we know the functionality is working on all platforms.

collinjackson commented:
@cqhchan -- Please let me know if you'd like to implement the iOS side of the pull request or if you want to abandon this PR. Thanks!

cqhchan commented:
hi @collinjackson , Im not too sure what IOS side u are referring to. The bug only appeared in Android. Regarding the additional feature to enable auto phone number verification, again it is an Android only feature in selected Android phones that support it

creativecreatorormaybenot commented:
@collinjackson Is this going to be merged soon then? (considering the comment)

collinjackson commented:
The API design in #1568 seems preferable to me, so I'm closing this one.
